run_custom_classifier.py

```diff
@@ -749,12 +751,20 @@ def metric_fn(per_example_loss, label_ids, logits, is_real_example):
                 predictions = tf.argmax(logits, axis=-1, output_type=tf.int32)
                 accuracy = tf.metrics.accuracy(
                     labels=label_ids, predictions=predictions, weights=is_real_example)
+                # add more metrics
+                pr, pr_op = tf.metrics.precision(
+                    labels=label_ids, predictions=predictions, weights=is_real_example)
+                re, re_op = tf.metrics.recall(
+                    labels=label_ids, predictions=predictions, weights=is_real_example)
+                f1 = (2 * pr * re) / (pr + re)  # f1-score for binary classification
                 loss = tf.metrics.mean(values=per_example_loss, weights=is_real_example)
                 return {
                     "eval_accuracy": accuracy,
-                    "eval_loss": loss,
+                    "eval_precision": (pr, pr_op),
+                    "eval_recall": (re, re_op),
+                    "eval_f1score": (f1, tf.identity(f1)),
```

Comment:
I tested it and found that several new metrics(precision/recall/f1) in Multi-GPU mode will cause ERROR `invalid type`. If you remove line 754-759, 763-765, it will go through the eval stage. This may be because `tf.metrics` is not compatible with `MirroredStrategy`. 
For more details, the bug here is the metrics like precision/recall/f1 was never updated to use the `_aggregate_across_towers()` wrapper that the non contrib `tf.metrics` use such as `tf.mean_per_class_accuracy`.
REF and Clues: https://github.com/tensorflow/tensorflow/issues/23172

@shishishu  Do you have any idea to solve the problem?

Reply:
Got it. I think the reason for recent failures may be related to these new metrics when used with multi-gpus.